fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -315,9 +316,14 @@ public Object getSubspaceKey() {
      * @see #getSubspaceKey()
      */
     public void setSubspaceKey(@Nonnull Object subspaceKey) {
+        usingExplicitSubspaceKey = true;
         this.subspaceKey = subspaceKey;
     }
 
+    public boolean hasExplicitSubspaceKey() {
+        return usingExplicitSubspaceKey;
```

Comment:
Probably could use a Javadoc comment.

There's some amount of weirdness with the contract of this method in that it's, like, "has a subspace key been set by someone", and that "someone" might be the user (where I think calling it an explicit subspace key is definitely fine), or that "someone" might be the `RecordMetaDataBuilder`, where "explicit" is maybe less sensical.

Reply:
Not really. I think documenting exactly what the contract is (which it looks like you did in a subsequent change) gets us most of the way there.